mautic/mautic
# Email Drafts

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#345
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | N/A



## Description
"Drafts" feature for email from Acquia. User documentation can be found at the link above.

User must enable this feature in the config:
![Screenshot 2024-11-19 at 11 50 54](https://github.com/user-attachments/assets/9a738244-c57a-4d2a-8bd0-63b16e8e476d)

---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Ensure email drafts are enabled by modifying the configuration as described above.
3. Create an email with some content, and save it.
4. Edit the email again, and make a change to the content
5. Instead of saving, use the "Save as Draft" button to save the draft.
6. Check both the Preview URL and the Draft Preview URL to see the differences between the emails.
7. Edit the email again, and apply the draft. See that the changes are applied and the Draft Preview URL is no longer present.
8. Edit the email again and create another draft.
9. Discard the draft this time, and see that the changes do not get applied.
